src/murmur/Cert.cpp

```diff
@@ -132,57 +232,62 @@ void Server::initializeCert() {
 
 	QString issuer;
 
-	QStringList issuerNames = qscCert.issuerInfo(QSslCertificate::CommonName);
-	if (!issuerNames.isEmpty()) {
-		issuer = issuerNames.first();
+	if (!qlCertificateChain.isEmpty()) {
+		QStringList issuerNames = qlCertificateChain[0].issuerInfo(QSslCertificate::CommonName);
+		if (!issuerNames.isEmpty()) {
+			issuer = issuerNames.first();
+		}
 	}
 
 	// Really old certs/keys are no good, throw them away so we can
 	// generate a new one below.
 	if (issuer == QString::fromUtf8("Murmur Autogenerated Certificate")) {
 		log("Old autogenerated certificate is unusable for registration, invalidating it");
-		qscCert = QSslCertificate();
-		qskKey  = QSslKey();
+		qlCertificateChain.clear();
+		qlCertificateChain << QSslCertificate();
+		qskKey = QSslKey();
 	}
 
 	// If we have a cert, and it's a self-signed one, but we're binding to
 	// all the same addresses as the Meta server is, use it's cert instead.
 	// This allows a self-signed certificate generated by Murmur to be
 	// replaced by a CA-signed certificate in the .ini file.
-	if (!qscCert.isNull() && issuer.startsWith(QString::fromUtf8("Murmur Autogenerated Certificate"))
-		&& !Meta::mp.qscCert.isNull() && !Meta::mp.qskKey.isNull() && (Meta::mp.qlBind == qlBind)) {
-		qscCert         = Meta::mp.qscCert;
-		qskKey          = Meta::mp.qskKey;
-		qlIntermediates = Meta::mp.qlIntermediates;
-
-		if (!qscCert.isNull() && !qskKey.isNull()) {
+	if (!qlCertificateChain.isEmpty() && !qlCertificateChain[0].isNull()
+		&& issuer.startsWith(QString::fromUtf8("Murmur Autogenerated Certificate"))
```

Comment:
```suggestion
		&& issuer.startsWith(QStringLiteral("Murmur Autogenerated Certificate"))
```

Reply:
That is not something I changed/introduced (just rearanged). Of course I can incorperate it in one of the commits but I would like to point out that `QString::fromUtf8("Murmur Autogenerated Certificate")` is used a few lines above (244), again. So should I change both occurrence or ignore this suggestion and leave it to an other commit/pr?

Reply:
Yes changing both in a separate commit would be best (provided the other use of `fromUtf8` is also to convert a String literal that doesn't contain any non-ASCII characters.